src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/CustomAttributeBuilder.cs

```diff
@@ -57,6 +57,8 @@ public CustomAttributeBuilder(ConstructorInfo con, object?[] constructorArgs, Pr
             ArgumentNullException.ThrowIfNull(namedFields);
             ArgumentNullException.ThrowIfNull(fieldValues);
 
+            AssemblyBuilder.EnsureDynamicCodeSupported();
```

Comment:
We only need this check in DefineDynamicAssembly.

We are working on adding back `AssemblyBuilderAccess.Save` option for Reflection.Emit (#15704). Having the check here would block `AssemblyBuilderAccess.Save` with native AOT. There is no reason to block it emitting assemblies to disk with native AOT.

Reply:
I added it in these places other than DynamicMethod and DefineDynamicAssembly because these were all the places we were throwing in native AOT. So I thought it would be consistent. For example:

https://github.com/dotnet/runtime/blob/7d91fd03c1e14b0a2097a8a39b4ec5ebb2b1c489/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Emit/CustomAttributeBuilder.cs#L6-L11

Reply:
Ok, we can leave these here for now. We can always delete these later as we make progress on RefEmit.Save.

cc @buyaa-n